### PR TITLE
nxcompext: Move code into Xserver subtree, as nxcompext requires Xser…

### PR DESCRIPTION
**Pull request by sunweaver**
…ver includes at build time.

Fixes ArcticaProject/nx-libs#276.

NOTE: This PR is marked as invalid / documentation for now, so people can see what I am working on.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -111,7 +109,7 @@ install-full:
 	sed -e 's|@@NXLIBDIR@@|$(NXLIBDIR)|g' bin/nxagent.in > bin/nxagent
 	$(INSTALL_PROGRAM) bin/nxagent $(DESTDIR)$(BINDIR)
 
-	for d in nxcompext nxcompshad; do \
+	for d in nxcompshad; do \
```
Comment (uli42) [inline Makefile]: no need to loop here anymore

File: nx-X11/programs/Xserver/Imakefile [hunk 2]
```diff
@@ -292,12 +293,17 @@ NXAGENTOBJS = hw/nxagent/miinitext.o     \
               dix/main.o
 #endif
 
+
```
Comment (uli42) [inline nx-X11/programs/Xserver/Imakefile]: strip that empty line

**Comment by sunweaver** [comment]
Just as a side note... This code segfaults at the moment. No clue, why.

**Comment by uli42** [comment]
Pgn.c can be renamed back to Png.c now.

**Comment by sunweaver** [comment]
@uli42: This branch has now become valid. Since @Ionic fixed the builds and commented out the libX11 symlink creation, this branch builds and works fine now.

@uli42: Are you available for reviewing? If yes, please go ahead for a second iteration. All above mentioned issues have already been considered with the current status of the PR branch.

**Comment by sunweaver** [comment]
We should discuss, what the best place for the former nxcompext files is. This PR places them into Xserver/hw/nxagent/. That's where they belong, I feel.

However, we may rename the files and prefix them with Compext<Name>.[ch] to denote that they stem from nxcompext. Such changes are more cosmetic and documentational rather than functional.

@Ionic: @vatral: @uli42: comments? feedback?

**Comment by uli42** [comment]
I can review that this evening, I think. But have a look at these for now:
1. SRC_COMPEXT occurs twice in the Imakefile
2. compext files should be moved to `hw/nxagent/(nx)compext`

Update: @sunweaver: I have reviewed it just now...

**Comment by uli42** [comment]
Please either remove those variables or add `_X_UNUSED`:

```
CompExt.c: In function ‘NXGetControlParameters’:
CompExt.c:824:18: warning: variable ‘req’ set but not used [-Wunused-but-set-variable]
   register xReq *req;
                  ^
CompExt.c: In function ‘NXGetFontParameters’:
CompExt.c:1172:37: warning: variable ‘req’ set but not used [-Wunused-but-set-variable]
   register xNXGetFontParametersReq *req;
                                     ^
CompExt.c: In function ‘NXCollectInputFocus’:
CompExt.c:4629:18: warning: variable ‘req’ set but not used [-Wunused-but-set-variable]
   register xReq *req;
                  ^
```

**Comment by uli42** [comment]
After replacing all pkgconfig lines for libjpeg by `-ljpeg` and reactivating my libX11-links stuff in the Imakefile it compiled fine. So please retry yourself without @Ionic's changes.

**Comment by sunweaver** [comment]
On  Mo 07 Nov 2016 13:23:54 CET, Ulrich Sibiller wrote:

> 1. SRC_COMPEXT occurs twice in the Imakefile

Und es muss auch noch SRCS_COMPEXT heißen... (plural). Die Dopplung  
ist beim rebasen passiert. Der Typo beim Tippen... ;-)

> 1. compext should be moved to a subdir

Hmmm... I am soooo lazy... How about prepending Compext... to related  
file names, e.g. `Compext<Name>.[ch]`?

## 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

**Comment by uli42** [comment]
And even too lazy for english ;-)

I think using a subdir is cleaner.

**Comment by sunweaver** [comment]
On  Mi 09 Nov 2016 17:55:48 CET, Ulrich Sibiller wrote:

> And even too lazy for english ;-)

Oops.

> I think using a subdir is cleaner.

See follow-up commit...  
https://github.com/sunweaver/nx-libs/commit/1d39e78eae398b1f233d5f3140145bc09d423d60

Mike

## 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

**Pull request merged**